### ISSUE TITLE
dataclass with ClassVar matches wrong regex

### ISSUE THREAD

**Issue by dseomn** [PyCQA/pylint#3405]
### Steps to reproduce
Run `pylint --rcfile=/dev/null --class-attribute-rgx='^x$' --attr-rgx='^y$'` on this file:

```python
import dataclasses
from typing import ClassVar

@dataclasses.dataclass
class Foo:
    x: ClassVar[int] = 0
```

### Current behavior
Pylint matches `x` against `attr-rgx` and raises `invalid-name`.

### Expected behavior
Pylint should match `x` against `class-attribute-rgx`, because of `ClassVar`: https://docs.python.org/3/library/dataclasses.html#class-variables

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Jan 19 2020, 22:34:52) 
[GCC 9.2.1 20200117]
```
